pkg/odo/cli/component/common_link.go

```diff
@@ -76,6 +145,40 @@ func (o *commonLinkOptions) complete(name string, cmd *cobra.Command, args []str
 }
 
 func (o *commonLinkOptions) validate(wait bool) (err error) {
+
+	if experimental.IsExperimentalModeEnabled() {
+		// let's validate if the service exists
+		_, err := svc.OperatorSvcExists(o.KClient, o.serviceType, o.serviceName)
+		if err != nil {
+			return err
+		}
+
+		// since the service exists, let's get more info to populate service binding request
+		// first get the CR itself
+		cr, err := o.KClient.GetCustomResource(o.serviceType)
+		if err != nil {
+			return err
+		}
+
+		// now get the group, version, kind information from CR
+		group, version, kind, err := svc.GetGVKFromCR(cr)
+		if err != nil {
+			return err
+		}
+
+		o.sbr.Spec.BackingServiceSelector = &sbo.BackingServiceSelector{
```

Comment:
should this be part of validate function? seems like something that should be present in Complete code?

Reply:
The reason why it's in `validate` is because we're validating a few lines above this if the service provided by the user actually exists. If it does, we populate the info about it, otherwise we don't. I think it's better in `validate` than `complete`.